### ISSUE TITLE
[#feature] `count: true` in criteria to return a count as opposed to a list.

### ISSUE THREAD

**Issue by lwansbrough** [balderdashy/sails#1951]
Currently I don't know how to return strictly a count of objects from a search/populate query. It would be really useful to be able to do `User.populate('followers', { count: true }).exec(cb)` and have Sails give me back something like `{ username: 'Loch', followers: 123 }`

AFAIK the only way of accomplishing this right now is to populate the association and count the length of the list, however this happens to be a great waste of RAM.

I can implement this and do a PR if you guys need more hands.

**Comment by mikermcneil** [comment]
Thanks for posting, @lwansbrough.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by tilomitra** [comment]
Any update on this? It's pretty important. Every ORM has something like this.

**Comment by RWOverdijk** [comment]
@tilomitra http://blog.spoonx.nl/sails-js-count-blueprint/

Then all you have to do is add a route with the blueprint.

You can also use something like this:

**CountController.js:**

```js
var actionUtil = require('sails/lib/hooks/blueprints/actionUtil');

module.exports = {
  count : function (req, res) {
    var Model    = sails.models[req.param('modelName')],
        criteria = actionUtil.parseCriteria(req);

    if (!Model) {
      return res.badRequest('invalid_parameter');
    }

    delete criteria.modelName;

    Model.count(criteria, function (error, response) {
      if (error) {
        return res.serverError('database_error', error);
      }

      res.ok({count: response});
    });
  }
};
```

**routes.js:**

`'GET /:modelName/count': 'CountController.count'`

**Comment by sercanov** [comment]
@RWOverdijk counting model is not a problem, the issue is about counting relations.

It's a really missing feature, I was trying to implement it myself via hooks but no chance by now.

**Comment by RWOverdijk** [comment]
@sercanov We had that same issue. We worked around it by reversing the search (doing a count on the child). If you wish to do a count with association restrictions (where clause) I'm afraid you're out of luck. You could write a custom query for that.

**Comment by sercanov** [comment]
I've been using @kristian-ackar 's count hook and tried to edit a bit to work with associated routes. But I'm kind a new in waterline / sails so I couldn't manage to finish it off. 

See the forked repo and changes, maybe someone could help 
https://github.com/sercanov/sails-hook-blueprint-count#associations-count-developments

**Comment by sercanov** [comment]
@tilomitra we've solved it with blueprint hook. Check it out 
https://github.com/sercanov/sails-hook-blueprint-utils

**Comment by akhoury** [comment]
if you're using Sails 1.0+ you can this is middleware i just wrote, instead of making 2 calls, a `X-Total-Count` will be automatically added in the response.

https://github.com/akhoury/sails-count-middleware
